src/plugins/intel_gpu/src/graph/impls/ocl/softmax.cpp

```diff
@@ -69,16 +69,31 @@ struct softmax_impl : typed_primitive_impl_ocl<softmax> {
 namespace detail {
 
 attach_softmax_impl::attach_softmax_impl() {
-    implementation_map<softmax>::add(impl_types::ocl, softmax_impl::create, {
-        std::make_tuple(data_types::f32, format::yxfb),
-        std::make_tuple(data_types::f16, format::yxfb),
-        std::make_tuple(data_types::f32, format::bfyx),
-        std::make_tuple(data_types::f16, format::bfyx),
-        std::make_tuple(data_types::f32, format::byxf),
-        std::make_tuple(data_types::f16, format::byxf),
-        std::make_tuple(data_types::f32, format::bfzyx),
-        std::make_tuple(data_types::f16, format::bfzyx),
-    });
+    const auto types = {data_types::f16, data_types::f32};
+    const auto formats = {
+            format::bfyx,
+            format::yxfb,
+            format::b_fs_yx_fsv16,
+            format::b_fs_yx_fsv32,
+            format::bs_fs_yx_bsv16_fsv16,
+            format::bs_fs_yx_bsv32_fsv16,
+            format::bs_fs_yx_bsv32_fsv32,
+            format::bfzyx,
+            format::b_fs_zyx_fsv16,
+            format::b_fs_zyx_fsv32,
+            format::bs_fs_zyx_bsv16_fsv32,
+            format::bs_fs_zyx_bsv16_fsv16,
+            format::bs_fs_zyx_bsv32_fsv32,
+            format::bs_fs_zyx_bsv32_fsv16
+    };
+
+    std::set<std::tuple<data_types, format::type>> keys;
+    for (const auto& t : types) {
+        for (const auto& f : formats) {
+            keys.emplace(t, f);
```

Comment:
Please use another `add` overload which will do the code above internally.

Reply:
Done.